useragent/mac/dnsfilter_ua_debugger_mac.sh

```diff
@@ -0,0 +1,91 @@
+  exit
+fi
+echo "======================================================" >> $ResultLog
+
+echo "Checking to see if 'DNSFilter Agent' is active..."
+echo "======================================================"
+if (sudo launchctl list | grep -q "com.dnsfilter.agent.macos.helper"); then
+  echo "DNSFilter Agent is active" >> $ResultLog
+else
+  echo "DNSFilter Agent is not active" >> $ResultLog
+fi
+
+echo "======================================================" >> $ResultLog
+
+# Check to see if "DNS Agent" is active
+echo "Checking to see if 'DNS Agent' is active..."
+if (sudo launchctl list | grep -q "io.netalerts.agent.macos.helper"); then
+  echo "DNS Agent is active" >> $ResultLog
+else 
+  echo "DNS Agent is not active" >> $ResultLog
+fi
+
+echo "======================================================" >> $ResultLog
+# Perform a ping for 4 attempts to 8.8.8.8
+echo "Performing ping"
+ping -c 4 8.8.8.8
+ping -c 4 8.8.8.8 >> $ResultLog
+
+echo "======================================================" >> $ResultLog
+
+# Perform nslookup -type=txt debug.dnsfilter.com
+echo "Performing first nslookup"
+nslookup -type=txt debug.dnsfilter.com
+nslookup -type=txt debug.dnsfilter.com >> $ResultLog
+
+echo "======================================================" >> $ResultLog
+
+# Perform nslookup -type=txt debug.dnsfilter.com 103.247.36.36
+echo "Performing second nslookup"
+nslookup -type=txt debug.dnsfilter.com 103.247.36.36
+nslookup -type=txt debug.dnsfilter.com 103.247.36.36 >> $ResultLog
+
+echo "======================================================" >> $ResultLog
+
+# Perform nslookup -type=txt debug.dnsfilter.com 103.247.37.37
+echo "Performing final nslookup"
+nslookup -type=txt debug.dnsfilter.com 103.247.37.37
+nslookup -type=txt debug.dnsfilter.com 103.247.37.37 >> $ResultLog
+
+
+echo "======================================================" >> $ResultLog
+echo "======================================================" 
+echo "======================================================" 
+# Gather Agent Logs from roaming client and move them to $HOME/Downloads/DNSFilterLogs
+sudo cp /var/log/com.dnsfilter.agent.macos.helper/daemon.log $HOME/Downloads/DNSFilterLogs
+sudo cp /var/log/io.netalerts.agent.macos.helper/daemon.log $HOME/Downloads/DNSFilterLogs
+echo "Agent Logs Copied"
+echo "Agent Logs Copied to DNSFilterLogs folder" >> $ResultLog
+echo "======================================================" >> $ResultLog
+
+
+#Check for port 53 bindings
+echo "Checking for any issues with localhost:53"
+echo "The follow PID and Applications are using port 53" >> $ResultLog
+lsof -i :53 | awk '{print $2, $1}' >> $ResultLog
+echo "==============================================" >> $ResultLog
+
+
+echo "======================================================" 
+echo "======================================================" 
+echo "The script has finished and the results are found in $ResultLog, please attach back to your support engineer"
```

Comment:
"The script is completed and the results are found in $ResultLog, please attach to your support ticket"

Reply:
As a bonus - could try executing `open $LogFolder`
If that is not flagged by security settings on mac, it would enhance usability because it opens the folder for them in the file manager.